gen3_tracker/meta/dataframer.py

```diff
@@ -842,121 +699,43 @@ def create_dataframe(
-    elif "valueSampledData" in resource_dict:
-        value = resource_dict["valueSampledData"]
-        value_normalized = value["data"]
-        value_source = "valueSampledData"
-    elif "valueTime" in resource_dict:
-        value_normalized = resource_dict["valueTime"]
-        value_source = "valueTime"
-    elif "valueDateTime" in resource_dict:
-        value_normalized = resource_dict["valueDateTime"]
-        value_source = "valueDateTime"
-    elif "valuePeriod" in resource_dict:
-        value = resource_dict["valuePeriod"]
-        value_normalized = f"{value['start']} to {value['end']}"
-        value_source = "valuePeriod"
-    # for debugging...
-    # else:
-    #     raise ValueError(f"value[x] not found in {resource_dict}")
-
-    return value_normalized, value_source
-
-
-def normalize_coding(resource_dict: dict) -> List[Tuple[str, str]]:
-    def extract_coding(coding_list):
-        # return a concatenated string
-        # return ','.join([coding.get('display', '') for coding in coding_list if 'display' in coding])
-        # or alternatively return an array
-        return [coding.get("display", coding.get("code", "")) for coding in coding_list]
-
-    def find_codings_in_dict(d: dict, parent_key: str = "") -> List[Tuple[str, str]]:
-        codings = []
-        for key, value in d.items():
-            if "valueCodeableConcept" in key:
-                continue
-            if isinstance(value, list):
-                # categories are values not codings in the pivot.
-                if "category" in key:
-                    continue
-                for item in value:
-                    if isinstance(item, dict):
-                        # valudeCodeableConcept is a value in the pivot not a coding
-                        # Check if the dict contains a 'coding' list
-                        if "coding" in item and isinstance(item["coding"], list):
-                            coding_string = extract_coding(item["coding"])
-                            codings.append((coding_string, key))
-                        # Recursively search in the dict
-                        codings.extend(find_codings_in_dict(item, key))
-            elif isinstance(value, dict):
-                # Check if the dict contains a 'coding' list
-                if "coding" in value and isinstance(value["coding"], list):
-                    coding_string = extract_coding(value["coding"])
-                    codings.append((coding_string, key))
-                # Recursively search in the dict
-                codings.extend(find_codings_in_dict(value, key))
-        return codings
-
-    return find_codings_in_dict(resource_dict)
-
-
 def is_number(s):
     """Returns True if string is a number."""
     try:
         int(s)
         return True
     except ValueError:
         return False
+    
+
+def get_subject(db: LocalFHIRDatabase, resource: dict) -> dict:
+    cursor = db.connect()
+    subject_key = resource['subject']['reference']
```

Comment:
Anywhere where there's dict lookup being done on a FHIR field that is not required there should be a get_nested_value function call instead so that an error is not raised

Reply:
updated